library/src/containers/Servers/ServerSecurity.tsx

```diff
@@ -8,82 +8,165 @@ import { ServerHelpers } from '../../helpers';
-    .map(requirement => {
-      const def: SecurityScheme =
-        securitySchemes[Object.keys(requirement.json())[0]];
-
-      if (!def) {
-        return null;
-      }
-      return <ServerSecurityItem securitySchema={def} key={def.type()} />;
-    })
-    .filter(Boolean);
-
-  if (!serverSecurities.length) {
-    return null;
-  }
+    if (protocol === 'kafka' || protocol === 'kafka-secure') {
+      renderedServerSecurities = (
+        <ServerSecurityItem protocol={protocol} securitySchema={null} key="" />
+      );
+    }
+  } else {
+    const serverSecurities: React.ReactNodeArray = serverRequirements
+      .map(requirement => {
+        const def: SecurityScheme =
+          securitySchemes[Object.keys(requirement.json())[0]];
 
-  return (
-    <div className="text-sm mt-4">
-      <h5 className="text-gray-700 text-base">Security:</h5>
+        if (!def) {
+          return null;
+        }
+        return (
+          <ServerSecurityItem
+            protocol={protocol}
+            securitySchema={def}
+            key={def.type()}
+          />
+        );
+      })
+      .filter(Boolean);
+    renderedServerSecurities = (
       <ul>
         {serverSecurities.map((security, idx) => (
           <li className="mt-2" key={idx}>
             {security}
           </li>
         ))}
       </ul>
+    );
+  }
+
+  return (
+    <div className="text-sm mt-4">
+      {renderedServerSecurities && (
+        <h5 className="text-gray-700 text-base">Security:</h5>
+      )}
+      {renderedServerSecurities}
     </div>
   );
 };
 
 interface ServerSecurityItemProps {
-  securitySchema: SecurityScheme;
+  securitySchema: SecurityScheme | null;
+  protocol: string;
 }
 
 const ServerSecurityItem: React.FunctionComponent<ServerSecurityItemProps> = ({
   securitySchema,
+  protocol,
 }) => {
   const schemas: React.ReactNodeArray = [];
-  if (securitySchema.name()) {
+  if (securitySchema && securitySchema.name()) {
     schemas.push(<span>Name: {securitySchema.name()}</span>);
   }
-  if (securitySchema.in()) {
+  if (securitySchema && securitySchema.in()) {
     schemas.push(<span>In: {securitySchema.in()}</span>);
   }
-  if (securitySchema.scheme()) {
+  if (securitySchema && securitySchema.scheme()) {
     schemas.push(<span>Scheme: {securitySchema.scheme()}</span>);
   }
-  if (securitySchema.bearerFormat()) {
+  if (securitySchema && securitySchema.bearerFormat()) {
     schemas.push(<span>Bearer format: {securitySchema.bearerFormat()}</span>);
   }
-  if (securitySchema.openIdConnectUrl()) {
+  if (securitySchema && securitySchema.openIdConnectUrl()) {
     schemas.push(
       <Href href={securitySchema.openIdConnectUrl()} className="underline">
         Connect URL
       </Href>,
     );
   }
 
-  const flows = securitySchema.flows();
+  let renderedKafkaSecurity;
+  if (protocol === 'kafka' || protocol === 'kafka-secure') {
+    let securityProtocol;
+    let saslMechanism;
```

Comment:
Logic for this `if` (only logic for retrieving values of `securityProtocol` and `saslMechanism`) should be written in https://github.com/asyncapi/asyncapi-react/blob/next/library/src/helpers/server.ts class as `retrieveKafkaSecurity` or `getKafkaSecurity` function, so you should end with something like:

```js
let renderedKafkaSecurity;
if (protocol === 'kafka' || protocol === 'kafka-secure') {
  const { securityProtocol, saslMechanism } = ServerHelpers.retrieveKafkaSecurity(protocol, securitySchema);
  
  renderedKafkaSecurity = ( ... );
}
```